src/slack_interaction_handler.ts

```diff
@@ -997,12 +1174,129 @@ export async function receiveResetDemo({
 
       // Store the relevant information in the modal so that when the requested action is confirmed
       // the data needed for the necessary actions is available.
-      slackView = SlackBlockUtil.resetConfirmationSlackView(
+      slackView = SlackBlockUtil.confirmationSlackView(
         SlackCallbackId.RESET_DEMO,
-        modalPrivateMetadata
+        modalPrivateMetadata,
+        'Are you sure you want to end your demo conversation with this volunteer?\n\nYou will no longer be able to send messages to or receive messages from them, and they will be treated as a new demo voter the next time they send a text to this phone number.'
+      );
+    }
+
+    await SlackApiUtil.updateModal(viewId, slackView);
+  } catch (e) {
+    // Update the modal to say that there was an error, then re-throw the
+    // error so it ends up in Sentry / the logs
+    await SlackApiUtil.updateModal(
+      viewId,
+      SlackBlockUtil.getErrorSlackView(
+        'internal_server_error',
+        'Sorry, something went wrong. Please try again.'
+      )
+    );
+    throw e;
+  }
+}
+
+// This function receives the initial request to route to journey
+// and responds by creating a modal populated with data needed
+// to route the voter to journey if the Slack user confirms.
+export async function receiveRouteToJourney({
+  payload,
+  redisClient,
+  modalPrivateMetadata,
+  twilioPhoneNumber,
+  userId,
+  viewId,
+}: {
+  payload: SlackInteractionEventPayload;
+  redisClient: PromisifiedRedisClient;
+  modalPrivateMetadata: SlackModalPrivateMetadata;
+  twilioPhoneNumber: string;
+  userId: string;
+  viewId: string;
+}): Promise<void> {
+  logger.info(`Entering SLACKINTERACTIONHANDLER.receiveRouteToJourney`);
+  let slackView;
+
+  try {
+    const redisUserInfoKey = `${userId}:${twilioPhoneNumber}`;
+    const userInfo = (await RedisApiUtil.getHash(
+      redisClient,
+      redisUserInfoKey
+    )) as UserInfo;
+
+    if (!userInfo) {
+      modalPrivateMetadata.success = false;
+      modalPrivateMetadata.failureReason = 'no_user_info';
+      await DbApiUtil.logCommandToDb(modalPrivateMetadata);
+      throw new Error(
+        `SLACKINTERACTIONHANDLER.receiveRouteToJourney: Interaction received for voter who has redisData but not userInfo: active redisData key is ${payload.channel.id}:${payload.message.ts}, userInfo key is ${redisUserInfoKey}.`
       );
     }
 
+    if (!(payload.channel.id === userInfo.activeChannelId)) {
```

Comment:
nit: probably easier to read as `payload.channel.id !== userInfo.activeChannelId`

Reply:
thanks!